AsyncDisplayKit/Details/ASPINRemoteImageDownloader.m

```diff
@@ -1,44 +1,114 @@
 //
 //  ASPINRemoteImageDownloader.m
-//  Pods
+//  AsyncDisplayKit
 //
 //  Created by Garrett Moon on 2/5/16.
-//
+//  Copyright © 2016 Facebook. All rights reserved.
 //
 
 #ifdef PIN_REMOTE_IMAGE
 #import "ASPINRemoteImageDownloader.h"
 
 #import "ASAssert.h"
 #import "ASThread.h"
+#import "ASImageContainerProtocolCategories.h"
+
+#if __has_include ("PINAnimatedImage.h")
+#define PIN_ANIMATED_AVAILABLE 1
+#import "PINAnimatedImage.h"
+#import <PINRemoteImage/PINAlternateRepresentationProvider.h>
+#else
+#define PIN_ANIMATED_AVAILABLE 0
+#endif
 
 #import <PINRemoteImage/PINRemoteImageManager.h>
+#import <PINRemoteImage/NSData+ImageDetectors.h>
 #import <PINCache/PINCache.h>
 
+#if PIN_ANIMATED_AVAILABLE
+@interface ASPINRemoteImageDownloader () <PINRemoteImageManagerAlternateRepresentationProvider>
+
+@end
+
+@interface PINAnimatedImage (ASPINRemoteImageDownloader) <ASAnimatedImageProtocol>
+
+@end
+
+@implementation PINAnimatedImage (ASPINRemoteImageDownloader)
```

Comment:
Is compilation works equivalently, I think it would be best to move this category implementation to beneath the primary @implementation block below.